### PR TITLE
Build in docker

### PR DESCRIPTION
**Pull request by christopherbryanboyd**
Hi @gamerson , here is the full docker build support, succeeding for me.
To use, please invoke `./gradlew buildInDocker` from the `liferay-blade-cli` project.

This will build the Dockerfile and run the resulting image, which by default invokes `clean build --scan`.

The commands to be run can be customized with a gradle project property, `-PgradleTasks` like this:
```
./gradlew buildInDocker -PgradleTasks="clean,jar,test,--debug-jvm"
```

### CODE REVIEW

**Comment by gamerson** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,14 @@
+FROM openjdk:8-stretch
+
+EXPOSE 5005
```
Comment (gamerson) [inline Dockerfile]: is this for debugging the gradle build?

File: build.gradle [hunk 2]
```diff
@@ -55,6 +56,28 @@ clean {
 	}
 }
 
+buildInDocker {
+	LinkedList commands = new LinkedList <String> ();
+
+	commands.addAll(['docker', 'build', '--tag', 'liferay-gradle', '.'])
```
Comment (gamerson) [inline build.gradle]: can we split this into two tasks?  one for building the image and another for running the container?

File: extensions/maven-profile/src/test/java/com/liferay/blade/extensions/maven/profile/BuildServiceCommandMavenTest.java [hunk 3]
```diff
@@ -81,7 +81,7 @@ public void testMavenServiceBuilder() throws Exception {
 		args = new String[] {"--base", mavenworkspace.getPath(), "buildService"};
 
 		BladeTestResults results = TestUtil.runBlade(mavenworkspace, _extensionsDir, args);
-
+
```
Comment (gamerson) [inline extensions/maven-profile/src/test/java/com/liferay/blade/extensions/maven/profile/BuildServiceCommandMavenTest.java]: whitespace

File: gradle.properties [hunk 4]
```diff
@@ -0,0 +1,3 @@
+org.gradle.daemon=true
+org.gradle.parallel=true
```
Comment (gamerson) [inline gradle.properties]: i'd rather leave the daemon off and parallel off for now.

File: build.gradle [hunk 5]
```diff
@@ -55,6 +56,28 @@ clean {
...
+
+	commandLine commands.toArray()
+
+	commands.clear()
+
+	commands.addAll(['docker', 'run', '--rm', '--init', '-p', '5004:5004', 'liferay-gradle'])
```
Comment (gamerson) [inline build.gradle]: up above you expose port 5005, why is it 5004 here?

File: build.gradle (hunk 2 above)
Comment (gamerson) [reply build.gradle]: Also we should give a tag that is `liferay-blade-cli-build`

File: Dockerfile (hunk 1 above)
Comment (christopherbryanboyd) [reply Dockerfile]: Yes correct

**Comment by gamerson** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by gamerson** [comment]
Pull request submitted to https://github.com/christopherbryanboyd/liferay-blade-cli/pull/27. [See changes here.](https://github.com/gamerson/liferay-blade-cli/compare/1df03a94~2...1df03a94) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Pull request closed without merging**